### PR TITLE
implement '#dark?' convenience method for dor-rights-auth objects

### LINKED ISSUE #37: add a dark? method to dor-rights-auth
convenience method addressing part of sul-dlss/argo#1464

### PR DESCRIPTION
**Pull request by jmartin-sul**
add a convenience method for asking whether an object's rightsMetadata renders it `dark` at the object level.

shore up tests, fix a minor bug that was discovered along the way.

closes #37

### CODE REVIEW

**Comment by jmartin-sul** [comment]
build is failing on ruby 2.1.2 (because it's not obtainable for download to the test VM).  ruby 2.1 support [ended over 2 years ago](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/).  do we still run it anywhere?  should we remove it from the build list?

will also fix the hound complaints.

**Comment by jcoyne** [comment]
@jmartin-sul I think everything we have is on ruby 2.5 or 2.6

File: lib/dor/rights_auth.rb [hunk 1]
```diff
@@ -49,6 +49,23 @@ def embargoed?
       @embargoed
     end
 
+    def check_index_elements_calculated!
+      unless index_elements.size > 0
```
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply lib/dor/rights_auth.rb]: if i use a guard clause, hound will complain about line length.

i could assign the message to a string first, and use that, but that feels overly circuitous.

happy to do whatever, if folks care, but i'm inclined to leave things as they are here.

File: lib/dor/rights_auth.rb [hunk 2]
```diff
@@ -277,7 +288,7 @@ def self.validate_lite(doc)
     def self.extract_index_terms(doc)
       terms = []
       machine = doc.at_xpath("//rightsMetadata/access[@type='read' and not(file)]/machine")
-      terms.push 'none_discover'  if doc.at_xpath("//rightsMetadata/access[@type='discover']/machine/none")
+      terms.push 'none_discover'  if doc.at_xpath("//rightsMetadata/access[@type='discover']/machine/none") || doc.at_xpath("//rightsMetadata/access[@type='discover']/machine[not(*)]")
```
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply lib/dor/rights_auth.rb]: considering the surrounding code, this actually feels like the most readable and consistent approach, unless we do some refactoring and cleanup of the existing code (which i'm not inclined to do in this PR).

**Comment by jmartin-sul** [comment]
> Can you fix the indentation issues?

done, hopefully

> I think everything we have is on ruby 2.5 or 2.6

removed 2.1.2 from the travis build


addressed (but didn't fix) all hound complaints.


@jcoyne, i think this is ready for another look.

**Pull request merged**